### ISSUE TITLE
Cannot use All the icons in dashboard anymore

### ISSUE THREAD

**Issue by liar666** [doomemacs/doomemacs#7455]
### I confirm that...

- [X] I have searched the [issue tracker](https://github.com/doomemacs/doomemacs/issues), [documentation](https://docs.doomemacs.org), [FAQ](https://docs.doomemacs.org/-/faq), [Discourse](https://discourse.doomemacs.org), and [Google](https://google.com), in case this issue has already been reported/resolved.

- [X] I have read "[How to Debug Issues](https://doomemacs.org/d/how2debug)", and will use it to provide as much information about this issue as possible.

- [X] The issue can be reproduced on the **latest** available commit of Doom.

- [X] The issue can be reproduced on a stable release of Emacs, such as 27, 28, or 29. *(Unstable versions end in .50, .60, or .9x)*

### Expected behavior

Expecting all-the-icons to work correctly, everywhere, included in Doom Dashboard

### Current behavior

I copyed some code from some forum to create a new entry in DoomEmacs' Dashboard (show "Open scratch buffer" with a pretty trashcan icon):
```
(add-to-list '+doom-dashboard-menu-sections
             '("Open *scratch* buffer"
               :icon (all-the-icons-octicon "trashcan" :face 'doom-dashboard-menu-title)
               :face (:inherit (doom-dashboard-menu-title bold))
               :action doom/open-scratch-buffer))
```

Worked like a charm until a few days ago, when I upgraded Doom.

Now when I start `emacs --debug-init` I see:
```
Debugger entered--Lisp error: (void-function all-the-icons-octicon)
  (all-the-icons-octicon "trashcan" :face 'doom-dashboard-menu-title)
  eval((all-the-icons-octicon "trashcan" :face 'doom-dashboard-menu-title) t)
  (if (stringp icon) icon (eval icon t))
  (let ((icon (if (stringp icon) icon (eval icon t)))) (format (format "%s%%s%%-10s" (if icon "%3s\11" "%3s")) (or icon "") (let ((temp-buffer (generate-new-buffer " *temp*" t))) (save-current-buffer (set-buffer temp-buffer) (unwind-protect (progn (insert-text-button label 'action (list ... ... ...) 'face (or face ...) 'follow-link t 'help-echo (format "%s (%s)" label ...)) (format "%-37s" (buffer-string))) (and (buffer-name temp-buffer) (kill-buffer temp-buffer))))) (propertize (or key (let* ((keymaps (and t ...)) (key (and keymaps ...))) (if key (let (...) (save-current-buffer ... ...)))) "") 'face 'doom-dashboard-menu-desc)))
  (+doom-dashboard--center (- +doom-dashboard--width 1) (let ((icon (if (stringp icon) icon (eval icon t)))) (format (format "%s%%s%%-10s" (if icon "%3s\11" "%3s")) (or icon "") (let ((temp-buffer (generate-new-buffer " *temp*" t))) (save-current-buffer (set-buffer temp-buffer) (unwind-protect (progn (insert-text-button label ... ... ... ... ... t ... ...) (format "%-37s" ...)) (and (buffer-name temp-buffer) (kill-buffer temp-buffer))))) (propertize (or key (let* ((keymaps ...) (key ...)) (if key (let ... ...))) "") 'face 'doom-dashboard-menu-desc))))
  (insert (+doom-dashboard--center (- +doom-dashboard--width 1) (let ((icon (if (stringp icon) icon (eval icon t)))) (format (format "%s%%s%%-10s" (if icon "%3s\11" "%3s")) (or icon "") (let ((temp-buffer (generate-new-buffer " *temp*" t))) (save-current-buffer (set-buffer temp-buffer) (unwind-protect (progn ... ...) (and ... ...)))) (propertize (or key (let* (... ...) (if key ...)) "") 'face 'doom-dashboard-menu-desc)))) (if (display-graphic-p) "\n\n" "\n"))
  (progn (insert (+doom-dashboard--center (- +doom-dashboard--width 1) (let ((icon (if (stringp icon) icon (eval icon t)))) (format (format "%s%%s%%-10s" (if icon "%3s\11" "%3s")) (or icon "") (let ((temp-buffer ...)) (save-current-buffer (set-buffer temp-buffer) (unwind-protect ... ...))) (propertize (or key (let* ... ...) "") 'face 'doom-dashboard-menu-desc)))) (if (display-graphic-p) "\n\n" "\n")))
  (if (and (fboundp action) (or (null when) (eval when t))) (progn (insert (+doom-dashboard--center (- +doom-dashboard--width 1) (let ((icon (if ... icon ...))) (format (format "%s%%s%%-10s" (if icon "%3s\11" "%3s")) (or icon "") (let (...) (save-current-buffer ... ...)) (propertize (or key ... "") 'face 'doom-dashboard-menu-desc)))) (if (display-graphic-p) "\n\n" "\n"))))
  (progn (let ((--cl-keys-- --cl-rest--)) (while --cl-keys-- (cond ((memq (car --cl-keys--) '(:icon :action :when :face :key :allow-other-keys)) (setq --cl-keys-- (cdr (cdr --cl-keys--)))) ((car (cdr (memq ... --cl-rest--))) (setq --cl-keys-- nil)) (t (error "Keyword argument %s not one of (:icon :action :whe..." (car --cl-keys--)))))) (if (and (fboundp action) (or (null when) (eval when t))) (progn (insert (+doom-dashboard--center (- +doom-dashboard--width 1) (let ((icon ...)) (format (format "%s%%s%%-10s" ...) (or icon "") (let ... ...) (propertize ... ... ...)))) (if (display-graphic-p) "\n\n" "\n")))))
  (let* ((--cl-rest-- section) (label (if --cl-rest-- (car-safe (prog1 --cl-rest-- (setq --cl-rest-- (cdr --cl-rest--)))) (signal 'wrong-number-of-arguments (list '(label &key icon action when face key) (length --cl-rest--))))) (icon (car (cdr (plist-member --cl-rest-- ':icon)))) (action (car (cdr (plist-member --cl-rest-- ':action)))) (when (car (cdr (plist-member --cl-rest-- ':when)))) (face (car (cdr (plist-member --cl-rest-- ':face)))) (key (car (cdr (plist-member --cl-rest-- ':key))))) (progn (let ((--cl-keys-- --cl-rest--)) (while --cl-keys-- (cond ((memq (car --cl-keys--) '...) (setq --cl-keys-- (cdr ...))) ((car (cdr ...)) (setq --cl-keys-- nil)) (t (error "Keyword argument %s not one of (:icon :action :whe..." (car --cl-keys--)))))) (if (and (fboundp action) (or (null when) (eval when t))) (progn (insert (+doom-dashboard--center (- +doom-dashboard--width 1) (let (...) (format ... ... ... ...))) (if (display-graphic-p) "\n\n" "\n"))))))
  (let ((section (car --dolist-tail--))) (let* ((--cl-rest-- section) (label (if --cl-rest-- (car-safe (prog1 --cl-rest-- (setq --cl-rest-- ...))) (signal 'wrong-number-of-arguments (list '... (length --cl-rest--))))) (icon (car (cdr (plist-member --cl-rest-- ':icon)))) (action (car (cdr (plist-member --cl-rest-- ':action)))) (when (car (cdr (plist-member --cl-rest-- ':when)))) (face (car (cdr (plist-member --cl-rest-- ':face)))) (key (car (cdr (plist-member --cl-rest-- ':key))))) (progn (let ((--cl-keys-- --cl-rest--)) (while --cl-keys-- (cond ((memq ... ...) (setq --cl-keys-- ...)) ((car ...) (setq --cl-keys-- nil)) (t (error "Keyword argument %s not one of (:icon :action :whe..." ...))))) (if (and (fboundp action) (or (null when) (eval when t))) (progn (insert (+doom-dashboard--center (- +doom-dashboard--width 1) (let ... ...)) (if (display-graphic-p) "\n\n" "\n")))))) (setq --dolist-tail-- (cdr --dolist-tail--)))
  (while --dolist-tail-- (let ((section (car --dolist-tail--))) (let* ((--cl-rest-- section) (label (if --cl-rest-- (car-safe (prog1 --cl-rest-- ...)) (signal 'wrong-number-of-arguments (list ... ...)))) (icon (car (cdr (plist-member --cl-rest-- ...)))) (action (car (cdr (plist-member --cl-rest-- ...)))) (when (car (cdr (plist-member --cl-rest-- ...)))) (face (car (cdr (plist-member --cl-rest-- ...)))) (key (car (cdr (plist-member --cl-rest-- ...))))) (progn (let ((--cl-keys-- --cl-rest--)) (while --cl-keys-- (cond (... ...) (... ...) (t ...)))) (if (and (fboundp action) (or (null when) (eval when t))) (progn (insert (+doom-dashboard--center ... ...) (if ... "\n\n" "\n")))))) (setq --dolist-tail-- (cdr --dolist-tail--))))
  (let ((--dolist-tail-- +doom-dashboard-menu-sections)) (while --dolist-tail-- (let ((section (car --dolist-tail--))) (let* ((--cl-rest-- section) (label (if --cl-rest-- (car-safe ...) (signal ... ...))) (icon (car (cdr ...))) (action (car (cdr ...))) (when (car (cdr ...))) (face (car (cdr ...))) (key (car (cdr ...)))) (progn (let ((--cl-keys-- --cl-rest--)) (while --cl-keys-- (cond ... ... ...))) (if (and (fboundp action) (or ... ...)) (progn (insert ... ...))))) (setq --dolist-tail-- (cdr --dolist-tail--)))))
  doom-dashboard-widget-shortmenu()
  run-hooks(+doom-dashboard-functions)
  (let ((pt (point))) (if (eq major-mode '+doom-dashboard-mode) nil (+doom-dashboard-mode)) (erase-buffer) (run-hooks '+doom-dashboard-functions) (goto-char pt) (+doom-dashboard-reposition-point-h))
  (progn (let ((pt (point))) (if (eq major-mode '+doom-dashboard-mode) nil (+doom-dashboard-mode)) (erase-buffer) (run-hooks '+doom-dashboard-functions) (goto-char pt) (+doom-dashboard-reposition-point-h)) (+doom-dashboard-resize-h) (+doom-dashboard--persp-detect-project-h) (+doom-dashboard-update-pwd-h) (current-buffer))
  (unwind-protect (progn (let ((pt (point))) (if (eq major-mode '+doom-dashboard-mode) nil (+doom-dashboard-mode)) (erase-buffer) (run-hooks '+doom-dashboard-functions) (goto-char pt) (+doom-dashboard-reposition-point-h)) (+doom-dashboard-resize-h) (+doom-dashboard--persp-detect-project-h) (+doom-dashboard-update-pwd-h) (current-buffer)) (if modified nil (restore-buffer-modified-p nil)))
  (let* ((modified (buffer-modified-p)) (buffer-undo-list t) (inhibit-read-only t) (inhibit-modification-hooks t)) (unwind-protect (progn (let ((pt (point))) (if (eq major-mode '+doom-dashboard-mode) nil (+doom-dashboard-mode)) (erase-buffer) (run-hooks '+doom-dashboard-functions) (goto-char pt) (+doom-dashboard-reposition-point-h)) (+doom-dashboard-resize-h) (+doom-dashboard--persp-detect-project-h) (+doom-dashboard-update-pwd-h) (current-buffer)) (if modified nil (restore-buffer-modified-p nil))))
  (save-current-buffer (set-buffer (doom-fallback-buffer)) (if doom-inhibit-log nil (doom--log "Reloading dashboard at %s" (format-time-string "%T"))) (let* ((modified (buffer-modified-p)) (buffer-undo-list t) (inhibit-read-only t) (inhibit-modification-hooks t)) (unwind-protect (progn (let ((pt (point))) (if (eq major-mode '+doom-dashboard-mode) nil (+doom-dashboard-mode)) (erase-buffer) (run-hooks '+doom-dashboard-functions) (goto-char pt) (+doom-dashboard-reposition-point-h)) (+doom-dashboard-resize-h) (+doom-dashboard--persp-detect-project-h) (+doom-dashboard-update-pwd-h) (current-buffer)) (if modified nil (restore-buffer-modified-p nil)))))
  (progn (save-current-buffer (set-buffer (doom-fallback-buffer)) (if doom-inhibit-log nil (doom--log "Reloading dashboard at %s" (format-time-string "%T"))) (let* ((modified (buffer-modified-p)) (buffer-undo-list t) (inhibit-read-only t) (inhibit-modification-hooks t)) (unwind-protect (progn (let ((pt ...)) (if (eq major-mode ...) nil (+doom-dashboard-mode)) (erase-buffer) (run-hooks '+doom-dashboard-functions) (goto-char pt) (+doom-dashboard-reposition-point-h)) (+doom-dashboard-resize-h) (+doom-dashboard--persp-detect-project-h) (+doom-dashboard-update-pwd-h) (current-buffer)) (if modified nil (restore-buffer-modified-p nil))))))
  (if (or (and (not +doom-dashboard-inhibit-refresh) (get-buffer-window (doom-fallback-buffer)) (not (window-minibuffer-p (frame-selected-window))) (not (run-hook-with-args-until-success '+doom-dashboard-inhibit-functions))) force) (progn (save-current-buffer (set-buffer (doom-fallback-buffer)) (if doom-inhibit-log nil (doom--log "Reloading dashboard at %s" (format-time-string "%T"))) (let* ((modified (buffer-modified-p)) (buffer-undo-list t) (inhibit-read-only t) (inhibit-modification-hooks t)) (unwind-protect (progn (let (...) (if ... nil ...) (erase-buffer) (run-hooks ...) (goto-char pt) (+doom-dashboard-reposition-point-h)) (+doom-dashboard-resize-h) (+doom-dashboard--persp-detect-project-h) (+doom-dashboard-update-pwd-h) (current-buffer)) (if modified nil (restore-buffer-modified-p nil)))))))
  +doom-dashboard-reload()
  (progn (set-window-buffer nil (doom-fallback-buffer)) (+doom-dashboard-reload))
  (if (equal (buffer-name) "*scratch*") (progn (set-window-buffer nil (doom-fallback-buffer)) (+doom-dashboard-reload)))
  (if noninteractive nil (setq doom-fallback-buffer-name +doom-dashboard-name initial-buffer-choice #'doom-fallback-buffer) (if fancy-splash-image nil (setq fancy-splash-image (expand-file-name +doom-dashboard-banner-file +doom-dashboard-banner-dir))) (if (equal (buffer-name) "*scratch*") (progn (set-window-buffer nil (doom-fallback-buffer)) (+doom-dashboard-reload))) (add-hook 'doom-load-theme-hook #'+doom-dashboard-reload-on-theme-change-h) (add-hook 'window-configuration-change-hook #'+doom-dashboard-resize-h) (add-hook 'window-size-change-functions #'+doom-dashboard-resize-h) (add-hook 'doom-switch-buffer-hook #'+doom-dashboard-reload-maybe-h) (add-hook 'delete-frame-functions #'+doom-dashboard-reload-frame-h) (add-hook 'persp-created-functions #'+doom-dashboard--persp-record-project-h) (add-hook 'persp-activated-functions #'+doom-dashboard--persp-detect-project-h) (if (daemonp) (progn (add-hook 'persp-activated-functions #'+doom-dashboard-reload-maybe-h))) (add-hook 'persp-before-switch-functions #'+doom-dashboard--persp-record-project-h))
  +doom-dashboard-init-h()
  funcall(+doom-dashboard-init-h)
  (condition-case e (funcall hook) ((debug error) (signal 'doom-hook-error (list hook e))))
  doom-run-hook(+doom-dashboard-init-h)
  run-hook-wrapped(doom-run-hook +doom-dashboard-init-h)
  (let ((doom--hook hook)) (run-hook-wrapped hook #'doom-run-hook))
  (condition-case e (let ((doom--hook hook)) (run-hook-wrapped hook #'doom-run-hook)) ((debug doom-hook-error) (if debug-on-error nil (lwarn hook :error "Error running hook %S because: %s" (if (symbolp (car (cdr e))) (symbol-name (car (cdr e))) (car (cdr e))) (car (cdr (cdr e))))) (signal 'doom-hook-error (cons hook (cdr e)))))
  (let ((hook (car --dolist-tail--))) (condition-case e (let ((doom--hook hook)) (run-hook-wrapped hook #'doom-run-hook)) ((debug doom-hook-error) (if debug-on-error nil (lwarn hook :error "Error running hook %S because: %s" (if (symbolp (car ...)) (symbol-name (car ...)) (car (cdr e))) (car (cdr (cdr e))))) (signal 'doom-hook-error (cons hook (cdr e))))) (setq --dolist-tail-- (cdr --dolist-tail--)))
  (while --dolist-tail-- (let ((hook (car --dolist-tail--))) (condition-case e (let ((doom--hook hook)) (run-hook-wrapped hook #'doom-run-hook)) ((debug doom-hook-error) (if debug-on-error nil (lwarn hook :error "Error running hook %S because: %s" (if (symbolp ...) (symbol-name ...) (car ...)) (car (cdr ...)))) (signal 'doom-hook-error (cons hook (cdr e))))) (setq --dolist-tail-- (cdr --dolist-tail--))))
  (let ((--dolist-tail-- hooks)) (while --dolist-tail-- (let ((hook (car --dolist-tail--))) (condition-case e (let ((doom--hook hook)) (run-hook-wrapped hook #'doom-run-hook)) ((debug doom-hook-error) (if debug-on-error nil (lwarn hook :error "Error running hook %S because: %s" (if ... ... ...) (car ...))) (signal 'doom-hook-error (cons hook (cdr e))))) (setq --dolist-tail-- (cdr --dolist-tail--)))))
  doom-run-hooks(doom-init-ui-hook)
  doom-init-ui-h()
  run-hooks(window-setup-hook)
  #f(compiled-function () #<bytecode -0xdde2dc30af9921a>)()
  normal-top-level()
Debugger entered--Lisp error: (void-function all-the-icons-octicon)
```
And the Dashboard buffer is empty.

But if I go to the scratch buffer and do `M-x all-the-icons-insert` and type "trash", it works perfectly...

So the fonts are installed and working perfectly. They are probably not loaded soon enough though...

Ho wcan we solve that?

### Steps to reproduce

1. Add the code above (to configure an entry in Dashboard with an icon) in your config.org
2. start emacs
3. see the error

### System Information

https://pastebin.com/xtN7ZCYS

**Comment by liar666** [comment]
Hi,

Thanks for the quick answer.

Sorry, I searched for "icons" in the issues and found a lot of issues. Unfortunately, I didn't find this entry, or I didn't find anything I thought useful in them.
 
One the thing is that I already had `nerd-icons` installed (on top of `all-the-icons`), so I didn't think most of the solutions in the  issues I read were relevant. 
I didn'rtunderstand why my config was not working, as the font is here. I realized just now that I try to get an icon for the specific `all-the-icons` font (I use `all-the-icons-octicon "trashcan"`), which might be the reason why it is not found, as I have to look into `nerd-font` now.
I tried to change that line to different things, both for the font and the name of the glyph (from what I see when I hit `nerd-icons-insert` and search for a glyph), but to no success:
- `nerd-icons-insert-octicon`
- `nerd-icons-octicon`
-  `nf-oct-tras` / `trash` / `trashcan` / `trashbin` / `nf-fa-trash`
``` 

I would appreciate some help or a pointer to a doc explaining how to switch from all-the-icon to nerd font glyphs/fonts names and commands, if you have any. Or on nerd-icons+Doom
